src/substrait/textplan/converter/HierarchyStack.h

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <google/protobuf/message.h>
```

Comment:
HierarchyStack isn't an external library so I'm not worried about exposing that symbol (it's only used by the converter which is all about protos).  Hiding the protobuf package from external callers isn't difficult though.  For instance here I can replace the header file with a forward declaration as I am not doing anything complex in the header file.  I do use this trick with Location so that users of Location do not need either protobuf or antlr4 as part of their includes:

`namespace google::protobuf {
  class Message;
} // namespace google::protobuf
`